src/Mvc/Mvc.Core/test/ModelBinding/Binders/DateTimeModelBinderTest.cs

```diff
@@ -0,0 +1,226 @@
+
+            // Assert
+            Assert.Equal(ModelBindingResult.Failed(), bindingContext.Result);
+            Assert.Empty(bindingContext.ModelState);
+        }
+
+        [Fact]
+        public async Task BindModel_NullableDatetime_ReturnsFailed_IfValueProviderEmpty()
+        {
+            // Arrange
+            var bindingContext = GetBindingContext(typeof(DateTime?));
+            var binder = GetBinder();
+
+            // Act
+            await binder.BindModelAsync(bindingContext);
+
+            // Assert
+            Assert.Equal(ModelBindingResult.Failed(), bindingContext.Result);
+            Assert.Empty(bindingContext.ModelState);
+        }
+
+        [Theory]
+        [InlineData("")]
+        [InlineData(" \t \r\n ")]
+        public async Task BindModel_CreatesError_IfTrimmedAttemptedValueIsEmpty_NonNullableDestination(string value)
+        {
+            // Arrange
+            var message = $"The value '{value}' is invalid.";
+            var bindingContext = GetBindingContext();
+            bindingContext.ValueProvider = new SimpleValueProvider
+            {
+                { "theModelName", value },
+            };
+            var binder = GetBinder();
+
+            // Act
+            await binder.BindModelAsync(bindingContext);
+
+            // Assert
+            Assert.False(bindingContext.Result.IsModelSet);
+            Assert.Null(bindingContext.Result.Model);
+
+            var error = Assert.Single(bindingContext.ModelState["theModelName"].Errors);
+            Assert.Equal(message, error.ErrorMessage, StringComparer.Ordinal);
+            Assert.Null(error.Exception);
+        }
+
+        [Theory]
+        [InlineData("")]
+        [InlineData(" \t \r\n ")]
+        public async Task BindModel_ReturnsNull_IfTrimmedAttemptedValueIsEmpty_NullableDestination(string value)
+        {
+            // Arrange
+            var bindingContext = GetBindingContext(typeof(DateTime?));
+            bindingContext.ValueProvider = new SimpleValueProvider
+            {
+                { "theModelName", value }
+            };
+            var binder = GetBinder();
+
+            // Act
+            await binder.BindModelAsync(bindingContext);
+
+            // Assert
+            Assert.Null(bindingContext.Result.Model);
+            var entry = Assert.Single(bindingContext.ModelState);
+            Assert.Equal("theModelName", entry.Key);
+        }
+
+        [Theory]
+        [InlineData(typeof(DateTime))]
+        [InlineData(typeof(DateTime?))]
+        public async Task BindModel_ReturnsModel_IfAttemptedValueIsValid(Type type)
+        {
+            // Arrange
+            var expected = new DateTime(2019, 06, 14, 2, 30, 4, 0, DateTimeKind.Utc);
+            var bindingContext = GetBindingContext(type);
+            bindingContext.ValueProvider = new SimpleValueProvider(new CultureInfo("fr-FR"))
+            {
+                { "theModelName", "2019-06-14T02:30:04.0Z" }
```

Comment:
nit: Suggest leaving out the fractional seconds or using the full round-trip format
```suggestion
                { "theModelName", "2019-06-14T02:30:04.0000000Z" }
```